[docs][menubar] Add Menubar component page (#47616)
mui-org/material-ui: docs/data/material/components/menubar/menubar.md

---
productId: material-ui
title: Menubar React component
components: Box, Button, Divider, List, ListItemButton, ListItemIcon, ListItemText, ListSubheader, Paper, Typography
---

# Menubar

<p class="description">A React component for building accessible application menus.</p>

{{"component": "@mui/docs/ComponentLinkHeader"}}

> The `menubar` [role](https://w3c.github.io/aria/#dfn-role) is used to create a menu bar similar to those found in Windows, Mac, and Gnome desktop applications. A menu bar provides a consistent set of frequently used commands, and its interaction should be similar to the typical menu bar interaction in a desktop graphical user interface.

Reviewer: ```suggestion
The menu bar is a visually persistent menu similar to those found in desktop applications that provides a consistent set of frequently used commands.
```
I think we can shorten this wording from the spec description, for our docs "Windows, Mac..." and "desktop graphical user interface" seem a bit redundant
